### PR TITLE
docs: move versioned docs to main

### PR DESCRIPTION
**Pull request by jeffomatic**
This PR moves historical versions of our guide/reference documentation into the main branch. Some of the tooling had to be adjusted so that SDK docs (which remain versioned using historical branches) are generated and uploaded correctly.

This makes the local file structure match the online structure more closely, and allows us to skip using backport PRs to move documentation updates from `main` into a historical branch. It also opens the door for documentation that lives outside the versioning system.

After this PR is merged, we should make sure to fold the Ivy Playground documentation into `docs/1.2` before docs are uploaded again.

### CODE REVIEW

**Comment by jeffomatic** [comment]
Most of the volume in this PR is in file copies--the ultimate change is not nearly as large as the file count would suggest. There is a clear narrative in the commit log of this branch.

File: cmd/docgenerate/main.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
+// Command docgenerate generates static documentation for Chain Core and the
+// Chain Core SDKs.
+//
+// Usage:
+//
+//     docgenerate chainPath outputDir
```
Comment (kr) [inline cmd/docgenerate/main.go]: Could/should this use `$CHAIN` instead of an arg?

**Comment by kr** [comment]
This generally looks pretty nice to me.

> Most of the volume in this PR is in file copies

Hah yeah looks like most of it is from Nitti SVG files?

File: cmd/docgenerate/main.go (hunk 1 above)
Comment (jeffomatic) [inline cmd/docgenerate/main.go]: I went back and forth on that myself. I think after a conversation we had earlier in the week about someone struggling with `build-cored-release`, it occurred to me that it was nice if a tool didn't have `$CHAIN` as a dependency if it didn't absolutely have to.

At any rate, the `upload-docs` script (which is probably how `docgenerate` is invoked 95% of the time) feeds `$CHAIN` as a parameter here, so I don't think we'll even have to worry about the extra verbosity in practice.

**Comment by jeffomatic** [comment]
> Hah yeah looks like most of it is from Nitti SVG files?

Yeah. Those are duplicated at the moment, but a future PR can start to make the different version dirs pull assets from a common location, which I think is one nice benefit of this change.

**Comment by kr** [comment]
LGTM

**Pull request merged**